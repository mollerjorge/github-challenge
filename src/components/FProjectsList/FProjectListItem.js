import React from "react";
import PropTypes from "prop-types";
import FProjectsListItemHolder from "./FProjectListItem.style.js";
import { Icon } from "antd";

const FProjectsListItem = ({
	project: { name, id, watchers_count },
	onClickProjectHandler
}) => {
	return (
		<FProjectsListItemHolder
			onClick={onClickProjectHandler.bind(this, id, name)}
		>
			<h4>{name}</h4>{" "}
			<div>
				<Icon type="eye" /> <span>{watchers_count}</span>
			</div>
		</FProjectsListItemHolder>
	);
};

FProjectsListItem.propTypes = {
	project: PropTypes.shape({
		name: PropTypes.string,
		id: PropTypes.number,
		watchers_count: PropTypes.number
	}),
	onClickProjectHandler: PropTypes.func.isRequired
};

export default FProjectsListItem;
