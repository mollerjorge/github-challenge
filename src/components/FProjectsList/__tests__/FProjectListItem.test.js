import {shallow} from 'enzyme';
import React from 'react';
import FProjectListItem from '../FProjectListItem';

describe('FProjectListItem', () => {
	const projectMock = {
		name: 'sample project',
		id: 1,
		watchers_count: 1000
	};
	const mockFn = jest.fn();
	const fprojectItem = shallow(<FProjectListItem project={projectMock} onClickProjectHandler={mockFn}/>);
	
	it('renders correctly', () => {
		expect(fprojectItem).toMatchSnapshot();
	});

	it('displays the name correctly', () => {
		expect(fprojectItem.find('h4').text()).toEqual('sample project');
	});

	it('displays the watchers_count correctly', () => {
		expect(fprojectItem.find('span').text()).toEqual('1000');
	});

	describe('user clicks on project', () => {

		beforeEach(() => {
			fprojectItem.find('FProjectsListItemHolder').simulate('click');
		});

		it('executes callback function once', () => {
			expect(mockFn).toHaveBeenCalledTimes(1);
		});

		it('executes calback function with proper args', () => {
			expect(mockFn).toHaveBeenCalledWith(1, 'sample project');
		});
		
	});


});