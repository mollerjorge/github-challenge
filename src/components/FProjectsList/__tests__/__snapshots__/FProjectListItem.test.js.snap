// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FProjectListItem renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FProjectsListItem
    onClickProjectHandler={[Function]}
    project={
        Object {
            "id": 1,
            "name": "sample project",
            "watchers_count": 1000,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <h4>
          sample project
</h4>,
        " ",
        <div>
          <Icon
                    type="eye"
          />
           
          <span>
                    1000
          </span>
</div>,
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "sample project",
        },
        "ref": null,
        "rendered": "sample project",
        "type": "h4",
      },
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Icon
              type="eye"
/>,
            " ",
            <span>
              1000
</span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "type": "eye",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 1000,
            },
            "ref": null,
            "rendered": 1000,
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <h4>
            sample project
</h4>,
          " ",
          <div>
            <Icon
                        type="eye"
            />
             
            <span>
                        1000
            </span>
</div>,
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "sample project",
          },
          "ref": null,
          "rendered": "sample project",
          "type": "h4",
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Icon
                type="eye"
/>,
              " ",
              <span>
                1000
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "type": "eye",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1000,
              },
              "ref": null,
              "rendered": 1000,
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
